trading/exchanges/rest_exchanges/rest_exchange.py

```diff
@@ -202,25 +212,17 @@ def get_client(self):
             self.logger.error(f"Order {order_id} failed to cancel | {e}")
         return False
 
-    # todo { 'type': 'trailing-stop' }
-    def create_order(self, order_type, symbol, quantity, price=None, stop_price=None):
+    async def create_order(self, order_type, symbol, quantity, price=None, stop_price=None):
         try:
-            if order_type == TraderOrderType.BUY_MARKET:
-                return self.client.create_market_buy_order(symbol, quantity)
-            elif order_type == TraderOrderType.BUY_LIMIT:
-                return self.client.create_limit_buy_order(symbol, quantity, price)
-            elif order_type == TraderOrderType.SELL_MARKET:
-                return self.client.create_market_sell_order(symbol, quantity)
-            elif order_type == TraderOrderType.SELL_LIMIT:
-                return self.client.create_limit_sell_order(symbol, quantity, price)
-            elif order_type == TraderOrderType.STOP_LOSS:
-                return None
-            elif order_type == TraderOrderType.STOP_LOSS_LIMIT:
-                return None
-            elif order_type == TraderOrderType.TAKE_PROFIT:
-                return None
-            elif order_type == TraderOrderType.TAKE_PROFIT_LIMIT:
-                return None
+            created_order = await self._create_specific_order(order_type, symbol, quantity, price)
+            # some exchanges are not returning the full order details on creation: fetch it if necessary
+            if created_order and not self._ensure_order_details_completeness(created_order):
+                if ecoc.ID.value in created_order:
+                    order_id = created_order[ecoc.ID.value]
+                    order_symbol = created_order[ecoc.SYMBOL.value] if ecoc.SYMBOL.value in created_order else None
+                    return await self.exchange_manager.get_exchange().get_order(order_id, order_symbol)
```

Comment:
Remove the local var and 
`return await self.exchange_manager.get_exchange().get_order(created_order[ecoc.ID.value], order_symbol)`

Reply:
Why that ? I think it makes it harder to read like this